src/common/valtext.cpp

```diff
@@ -332,6 +312,56 @@ void wxTextValidator::OnChar(wxKeyEvent& event)
     event.Skip(false);
 }
 
+bool wxTextValidator::IsValid(const wxUniChar& c) const
+{
+    if ( !m_validatorStyle ) // no filtering if HasFlag(wxFILTER_NONE)
+        return true;
+
+    if ( HasFlag(wxFILTER_SPACE) && wxIsspace(c) )
+        return true;
+
+    // Notice that if the same char is mistakenly present in both the
+    // include char list and exclude char list than the char will be
+    // excluded and a warning is given in debug build.
+
+    if ( IsCharExcluded(c) )
+        return false;
+
+    if ( IsCharIncluded(c) )
+        return true;
+
+    static const long mask = wxFILTER_ASCII|wxFILTER_ALPHA|wxFILTER_NUMERIC|
+                      wxFILTER_ALPHANUMERIC|wxFILTER_DIGITS|wxFILTER_XDIGITS;
+    
+    long flags = m_validatorStyle & mask;
+
+    if ( !flags )
+        // accept whatever this character is.
+        return true;
+
+    // Turn off the wxFILTER_XXX if the corresponding check failed...
```

Comment:
This comment needs to be made more explicit and state that the goal here is to check if any flags remain set at the end, otherwise it's really not clear what does this code do.